pkg/controllers/leaderworkerset_controller.go

```diff
@@ -131,6 +131,7 @@ func (r *LeaderWorkerSetReconciler) createHeadlessServiceIfNotExists(ctx context
 				Selector: map[string]string{
 					leaderworkerset.SetNameLabelKey: lws.Name,
 				},
+				PublishNotReadyAddresses: lws.Spec.PublishNotReadyAddresses,
```

Comment:
could you add a test case in the integration/e2e test that verifies the field? thanks

Reply:
Added an integration test. 

I don't know if we need a e2e test here.

Reply:
k, sg